lib/honeybadger/notification_subscriber.rb

```diff
@@ -21,7 +44,27 @@ def finish(name, id, payload)
     end
 
     def record(name, payload)
-      Honeybadger.event(name, payload)
+      if Honeybadger.config.load_plugin_insights_events?(:rails)
+        Honeybadger.event(name, payload)
+      end
+
+      if Honeybadger.config.load_plugin_insights_metrics?(:rails) && (metrics = find_metrics(name, payload))
+        metric_source 'rails'
+        metrics.each do |metric|
+          public_send(
+            metric.type,
+            [metric.value_key, metric.event].join('.'),
+            value: payload[metric.value_key],
+            **payload.slice(*metric.context)
+          )
+        end
+      end
+    end
+
+    def find_metrics(name, payload)
+      RAILS_METRICS.select do |metric|
+        metric.event.to_s == name.to_s && payload.keys.include?(metric.value_key) && (payload.keys & metric.context).any?
+      end
     end
```

Comment:
I'm nervous about the load that will be added by going through this loop for every event. Perhaps this could be optimized by converting RAILS_METRICS into a hash, keyed by name, and the value is a proc or something callable that accepts the payload and either returns an array of metrics to the caller (like the method currently does) or even calls `public_send` itself. Maybe that "something callable" is a Metric class that's more than just a struct. :)